File: backoffice/FondaBeckEndLogic/RestaurantsSearch/GetZonesCommand.cs
using System;
using com.ds201625.fonda.DataAccess.InterfaceDAO;
using com.ds201625.fonda.Domain;
using System.Collections.Generic;
using com.ds201625.fonda.BackEndLogic.Exceptions;

namespace com.ds201625.fonda.BackEndLogic
{
	/// <summary>
	/// Comando para obtener la lista de zonas
	/// con almenos un restaurant.
	/// </summary>
	public class GetZonesCommand : BaseCommand
	{
		public GetZonesCommand () : base () { }

		/// <summary>
		/// Inicializa los parametros.
		/// </summary>
		/// <returns>The parameters.</returns>
		protected override Parameter[] InitParameters ()
		{
			// recive 3 parametros
			Parameter[] parameter = new Parameter[3];

			// [0] String de busqueda.
			parameter [0] = new Parameter (typeof(string), false);

			// [1] limite de resultados.
			parameter [1] = new Parameter (typeof(int), false);

			// [2] Pagina de resultados.
			parameter [2] = new Parameter (typeof(int), false);

			return parameter;
		}

		/// <summary>
		/// Ejecucion del parametro.
		/// </summary>
		protected override void Invoke()
		{

			string q = (string) GetParameter (0);
			int max = -1;
			int page = 1;

			if (GetParameter (1) != null)
				max = (int) GetParameter (1);
			if (GetParameter (2) != null)
				page = (int) GetParameter (2);

			IZoneDAO zoneDao = FacDao.GetZoneDAO ();
			IList<Zone> zones = null;

			try {
				zones = zoneDao.FindAllWithRestaurants (q, max, page);
			} 
			catch (Exception e)
			{
				throw new FondaBackendLogicException ("Error al obtener lista de zonas", e);
			}

			Result = zones;
		}
	}
}

File: frontoffice/app/src/main/java/com/ds201625/fonda/logic/Commands/RestaurantCommands/GetZonesCommand.java
package com.ds201625.fonda.logic.Commands.RestaurantCommands;

import android.util.Log;

import com.ds201625.fonda.data_access.factory.FondaServiceFactory;
import com.ds201625.fonda.data_access.services.RestaurantService;
import com.ds201625.fonda.domains.Zone;
import com.ds201625.fonda.logic.BaseCommand;
import com.ds201625.fonda.logic.Parameter;

import java.util.List;

/**
 * Comando para obtener zonas
 */
public class GetZonesCommand extends BaseCommand {

    private String TAG = "GetZonesCommand";

    /**
     * Asignacion de los parametros del comando.
     * Parametro en posicion 0: Query
     * Parametro en posicion 1: Max
     * Pareametro en posicion 2: Page
     */
    @Override
    protected Parameter[] setParameters() {
        Parameter [] parameters = new Parameter[3];
        parameters[0] = new Parameter(String.class, false);
        parameters[1] = new Parameter(Integer.class, false);
        parameters[2] = new Parameter(Integer.class, false);

        return parameters;
    }

    @Override
    public void invoke () {
        Log.d(TAG, "Comando para obtener la lista de Categorias");
        List<Zone> zones = null;

        RestaurantService resService = FondaServiceFactory.getInstance()
                .getRestaurantService();

        int max = 0;
        int page = 0;
        try {

            if (getParameter(1) != null) {
                max = (int) getParameter(1);
            }
            if (getParameter(2) != null) {
                page = (int) getParameter(2);
            }
            String query = (String) getParameter(0);

            zones = resService.getZones(query, max, page);
        }
        catch (Exception e) {
            Log.e(TAG, "Se ha generado error en invoke al obtener las zonas de restaurantes", e);
            e.printStackTrace();
        }

        setResult(zones);
    }

}
